### DISCUSSION TITLE
How to Publish iOS/MacCatalyst in Windows

### DISCUSSION THREAD

**Discussion by jbtdevgit** [unoplatform/uno#13271] [Q&A] [answered]
Hi,

Need some help on how to publish on iOS/Mac Catalyst. As of now I am trying to publish on Mac Catalyst on `Visual Studio Mac 17.6.3` but I am getting an issue building similar to this https://github.com/dotnet/runtime/issues/90032 and currently using `.net6-0`. Is it possible to publish through Visual Studio on Windows like https://learn.microsoft.com/en-us/dotnet/maui/mac-catalyst/deployment/publish-app-store#publish-using-the-command-line .NET MAUI with just a CLI? 

I am currently using the following command `dotnet publish -f net6.0-maccatalyst -c Release -p:ServerAddress=address -p:ServerUser=username -p:ServerPassword=password -p:TcpPort=58181 -p:_DotNetRootRemoteDirectory=directory` but I am getting an error of `Specify which project or solution file to use because this folder contains more than one project or solution file.` 

Any insights would be helpful. Thanks!

**Comment by jeromelaban** [comment]
Thanks for the question! This particular message happens when trying to build at the root folder of the solution. You'll need to run this command in the `.Mobile` folder.

**Reply by jbtdevgit** [reply]
Hi @jeromelaban ,
Unfortunately it does not work as well, or am I missing something?
![unable_to_publish](https://github.com/unoplatform/uno/assets/44351076/b3a2180b-d7bb-4939-bac9-52544ae583c3)

**Reply by jbtdevgit** [reply]
I deleted .bak project file and it worked, but I am not getting the package file created, instead it is creating the .mobile.dll file 
![empty_pkg](https://github.com/unoplatform/uno/assets/44351076/3ec0517e-7dc2-4571-bff4-1ca520c8e8ae)

**Reply by jbtdevgit** [reply]
Tried `dotnet publish -f net6.0-maccatalyst -c Release` on visual studio mac and it worked, any reasons why it is not working on windows? I never wanna do this cause it is a bit slow and mac freezes a bit..

**Reply by jeromelaban** [reply]
Thanks for the update. This is very curious, because dotnet build/publish takes the csproj directly in the current folder. can you try specifying the csproj explicitly in the command line?

**Reply by jbtdevgit** [reply]
I did try this and same thing happened.

**Reply by jeromelaban** [reply]
Thanks. Can you show the command line that you used?

**Reply by jbtdevgit** [reply]
`dotnet publish MyProject.Mobile -f:net6.0-maccatalyst -c:Release /p:ServerAddress=address /p:ServerUser=username /p:ServerPassword=password /p:TcpPort=58181 /p:_DotNetRootRemoteDirectory=directory`

**Reply by jeromelaban** [reply]
Thanks. Can you try specifying explicitly `MyProject.Mobile.csproj`? If you keep getting this message, it means that dotnet build detects multiple csproj or sln files in its current context.

**Reply by jbtdevgit** [reply]
Hi @jeromelaban , I've tried this as well, and removed any csproj under the .Mobile folder. I can do this pretty much on the mac mini (without the server address, password, tcpport, and directory), but not on Windows.

**Reply by jeromelaban** [reply]
Thanks. Could you take a screenshot of the folder that contains the `.mobile` csproj? This is a very curious issue. In all cases, publish for mac catalyst is best on the mac directly, on a CI server preferably.

**Reply by jbtdevgit** [reply]
@jeromelaban 
![image](https://github.com/unoplatform/uno/assets/44351076/c78f6725-7f73-4030-9558-088a2c0fc42b)
Hmm, so as of now it is not possible to publish in windows for mac catalyst yea?

**Reply by jeromelaban** [reply, marked as answer]
That's correct, Catalyst apps cannot be published from Windows.
